File: python/leetcode/701. Insert into a Binary Search Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None


class Solution:
    def insertIntoBST(self, root: TreeNode, val: int) -> TreeNode:
        cur = root

        while cur:
            if val > cur.val:
                #right
                if cur.right is None:
                    cur.right = TreeNode(val)
                    return root
                cur = cur.right

            if val < cur.val:
                #left
                if cur.left is None:
                    cur.left = TreeNode(val)
                    return root
                cur = cur.left
        return root

File: javascript/leetcode/701. Insert into a Binary Search Tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} val
 * @return {TreeNode}
 */
var insertIntoBST = function(root, val) {
  let cur = root;

  while (cur) {
    if (val > cur.val) {
      if (!cur.right) {
        cur.right = new TreeNode(val);
        return root;
      }
      cur = cur.right;
    }

    if (val < cur.val) {
      if (!cur.left) {
        cur.left = new TreeNode(val);
        return root;
      }
      cur = cur.left;
    }
  }

  return root;
};
